index.d.ts

```diff
@@ -0,0 +1,20 @@
+/**
+Negative array index support `array[-1]` using [ES2015 `Proxy`](https://ponyfoo.com/articles/es6-proxies-in-depth).
+
+@example
+```
+import negativeArray = require('negative-array');
+
+// Adds negative array index support to any given array
+const unicorn = negativeArray(['🐴', '🎂', '🌈']);
+
+// Get the last item by using a negative index
+console.log(unicorn[-1]);
+//=> '🌈'
+
+// OMG, YES!
+```
+*/
+declare function negativeArray<T extends readonly unknown[]>(array: T): T;
```

Comment:
Does this cause the returned array to be `readonly` too? If yes, I'm not sure it's worth it, as we decided it's not worth returning `readonly` array as it's inconvenient for the user.

Reply:
No, this is not what's happening here. This is just a base type, this means that if the incoming type is an `ReadonlyArray` then the result will also be one, if the incoming type is a plain `Array` then the resulting type will be an `Array`.

This is even covered by the tests.

Reply:
Ah, sorry, I should have looked at the tests... But thanks for the explanation. I was not sure how the `readonly` keyword worked with the base type.